contracts/SaleShare.sol

```diff
@@ -144,6 +141,17 @@ contract SaleShare is IERC677Receiver, Ownable, ReentrancyGuard {
         return (userInfo.amount, userInfo.claimed, owed, claimable, refund);
     }
 
+    function getAllUserInfo() external view returns (uint[2][] memory) {
+      uint[2][] memory allUserData = new uint[2][](totalUsers);
+
+      for (uint i = 0; i < totalUsers; i++) {
+        allUserData[i][0] = userInfos[userIndex[i]].amount;
+        allUserData[i][1] = userInfos[userIndex[i]].score;
+      }
+
+      return allUserData;
+    }
```

Comment:
This might work but most RPC nodes / providers do have a cap on the amount of gas a transaction / queries can use, so just to be safe, in case there's a lot of participants, we should make this method paginated (pass in a page and pageSize).

That way we can get all participants in multiple smaller batches, see https://github.com/Thorstarter/thorstarter-contracts/blob/main/contracts/ForgeV1.sol#L157